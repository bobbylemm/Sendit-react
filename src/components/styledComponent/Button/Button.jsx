import styled from 'styled-components';

const Button = styled.button`
  background: ${props => (props.primary ? '#5680E9' : '#3AAFA9')};
  border-radius: 0.3rem;
  text-align: center;
  color: #fff;
  font-weight: bold;
  cursor: pointer;
  border: ${props => (props.primary ? '0.1rem solid #5680E9' : '0.1rem solid #3AAFA9')};
  margin: 1rem;
  padding: ${props => (props.small ? '0.4rem 0.5rem' : '1.2rem 0.5rem')};
  box-shadow: -0.1rem 0.3rem 0.5rem 0rem rgba(0, 0, 0, 0.66);
  text-transform: uppercase;
  transition: all 0.3s;
  display: ${props => (props.inlineButton ? 'inline-block' : 'block')};
`;

export default Button;
