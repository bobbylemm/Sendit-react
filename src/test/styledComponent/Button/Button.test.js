import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from '../../../components/styledComponent/Button';

describe('testing the button element', () => {
  it('should mount button', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', '#fff');
  });

  it('should apply styles according to passed props', () => {
    const tree = renderer.create(<Button primary small inlineButton />).toJSON();
    expect(tree).toHaveStyleRule('background', expect.stringContaining('#5680E9'));
    expect(tree).toHaveStyleRule('border', expect.stringContaining('0.1rem solid #5680E9'));
    expect(tree).toHaveStyleRule('padding', expect.stringContaining('0.4rem 0.5rem'));
    expect(tree).toHaveStyleRule('display', expect.stringContaining('inline-block'));
  });
});
