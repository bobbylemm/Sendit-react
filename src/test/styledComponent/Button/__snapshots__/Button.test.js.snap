// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the button element should mount button 1`] = `
.c0 {
  background: #3AAFA9;
  border-radius: 0.3rem;
  text-align: center;
  color: #fff;
  font-weight: bold;
  cursor: pointer;
  border: 0.1rem solid #3AAFA9;
  margin: 1rem;
  padding: 1.2rem 0.5rem;
  box-shadow: -0.1rem 0.3rem 0.5rem 0rem rgba(0,0,0,0.66);
  text-transform: uppercase;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  display: block;
}

<button
  className="c0"
/>
`;
